lib/liquid.lua

```diff
@@ -98,6 +102,41 @@ local ENDRAW = "ENDRAW"
 --EOF
 local EOF = 'EOF'
 --
+--
+
+--- When replace_first filter was used, all the given chars should scaped at
+--lua way, no regexp option. To avoid future issues with different liquid
+--engines from other languages, this function sanitize the string to replace in
+--the correct way
+--Function is using bytes instead of chars, because comparision is faster in
+--this way. All was tested that is jit compatible to make it faster, no NYI
+--found
+--More information:
+--https://issues.redhat.com/browse/THREESCALE-4937
+--https://www.lua.org/pil/20.2.html
+function sanitize_replace(str)
+  local tbl = {stringbyte(str, 1, #str)}
+  result = {}
+  for _, v in pairs(tbl) do
+    if v == 40 or -- byte for char (
```

Comment:
You could also store these values in a hashmap to avoid these comparisons on every call.